pkg/apis/core/v1alpha1/helper/helper.go

```diff
@@ -545,6 +550,11 @@ func ShootWantsBasicAuthentication(shoot *gardencorev1alpha1.Shoot) bool {
 	return *kubeAPIServerConfig.EnableBasicAuthentication
 }
 
+// ShootUsesUnmanagedDNS returns true if the shoot's DNS section is marked as 'unmanaged'.
+func ShootUsesUnmanagedDNS(shoot *gardencorev1alpha1.Shoot) bool {
+	return shoot.Spec.DNS != nil && len(shoot.Spec.DNS.Providers) > 0 && shoot.Spec.DNS.Providers[0].Type != nil && *shoot.Spec.DNS.Providers[0].Type == "unmanaged"
```

Comment:
Can we also use a constant here?
```suggestion
	return shoot.Spec.DNS != nil && len(shoot.Spec.DNS.Providers) > 0 && shoot.Spec.DNS.Providers[0].Type != nil && *shoot.Spec.DNS.Providers[0].Type == gardencorev1alpha1.DNSUnmanaged
```